### PR TITLE
feature(misc): introduce eachAsync

### PR DESCRIPTION
**Pull request by sberthier**
Based on https://github.com/BEE-BUZZINESS/f-promise-async/pull/14

Similar to bluebird.each, sequentially iterate over array with async iteratee function

@tchambard

### CODE REVIEW

File: src/lib/misc/each.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+/// !doc
+
+/// ## eachAsync
+/// ```ts
+/// eachAsync<T, O>(iterable: AsyncIterable<Awaited<T>>, iteratee: ArrayIterator<Awaited<T>>): Promise<void>
+/// ```
+/// Sequentially iterate over (`array`, `Promise of array` or `array of Promises`) elements using iteratee function.
+/// The iteratee function can return true to stop iterate.
+/// ```ts
+/// await each(Promise.resolve([0, 1, 2]), (value, index) => {
```
Comment (blocosted) [inline src/lib/misc/each.ts]: eachAsync

File: src/lib/misc/each.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+## eachAsync
+```ts
+eachAsync<T, O>(iterable: AsyncIterable<Awaited<T>>, iteratee: ArrayIterator<Awaited<T>>): Promise<void>
+```
+Sequentially iterate over (`array`, `Promise of array` or `array of Promises`) elements using iteratee function.
+The iteratee function can return true to stop iterate.
+```ts
+await each(Promise.resolve([0, 1, 2]), (value, index) => {
```
Comment (blocosted) [inline src/lib/misc/each.md]: eachAsync

File: src/lib/misc/each.ts (hunk 1 above)
Comment (sberthier) [reply src/lib/misc/each.ts]: done

File: src/lib/misc/each.md (hunk 2 above)
Comment (sberthier) [reply src/lib/misc/each.md]: done

**Review by blocosted — APPROVED** [review]

**Pull request merged**